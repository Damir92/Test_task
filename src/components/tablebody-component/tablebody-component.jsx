import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {makeStyles} from '@material-ui/core';

import {Columns} from '../../const';
import {getSortedUsers} from '../../store/selectors';

const useStyles = makeStyles(() => ({
  row: {
    borderBottom: `1px solid lightgray`,
  },
  cell: {
    padding: `3px 5px`,
    textAlign: `center`,
  },
}));

const TableBodyComponent = ({users}) => {
  const classes = useStyles();

  return (
    <tbody>
      {users.map((row, index) => (
        <tr
          key={`${row.id}-${index}`}
          className={classes.row}
        >
          {Columns.map((item) => (
            <td
              key={item.key}
              className={classes.cell}
            >
              {row[item.key]}
            </td>
          ))}
        </tr>
      ))}
    </tbody>
  );
};

TableBodyComponent.propTypes = {
  users: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string.isRequired,
        firstName: PropTypes.string.isRequired,
        lastName: PropTypes.string.isRequired,
        group: PropTypes.string,
      }).isRequired,
  ),
};

const mapStateToProps = (state) => ({
  users: getSortedUsers(state),
});

export {TableBodyComponent};
export default connect(mapStateToProps)(TableBodyComponent);
