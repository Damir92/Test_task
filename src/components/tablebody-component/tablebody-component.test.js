import React from 'react';
import {shallow} from 'enzyme';

import {TableBodyComponent} from './tablebody-component';

const users = [
  {
    id: `1`,
    firstName: `userName-1`,
    lastName: `userLastName-1`,
    group: ``,
  },
  {
    id: `2`,
    firstName: `userName-2`,
    lastName: `userLastName-2`,
    group: `user`,
  }
];

describe(`TableBody component`, () => {
  it(`TableBodyComponent snapshot`, () => {
    const tree = shallow(
        <TableBodyComponent
          users={users}
        />
    );

    expect(tree).toMatchSnapshot();
  });
});
