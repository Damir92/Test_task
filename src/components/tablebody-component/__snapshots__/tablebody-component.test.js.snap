// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableBody component TableBodyComponent snapshot 1`] = `
<tbody>
  <tr
    className="makeStyles-row-1"
    key="1-0"
  >
    <td
      className="makeStyles-cell-2"
      key="firstName"
    >
      userName-1
    </td>
    <td
      className="makeStyles-cell-2"
      key="lastName"
    >
      userLastName-1
    </td>
    <td
      className="makeStyles-cell-2"
      key="group"
    />
  </tr>
  <tr
    className="makeStyles-row-1"
    key="2-1"
  >
    <td
      className="makeStyles-cell-2"
      key="firstName"
    >
      userName-2
    </td>
    <td
      className="makeStyles-cell-2"
      key="lastName"
    >
      userLastName-2
    </td>
    <td
      className="makeStyles-cell-2"
      key="group"
    >
      user
    </td>
  </tr>
</tbody>
`;
